pkg/frontend/admin_ovsflows_get.go

```diff
@@ -0,0 +1,195 @@
+package frontend
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/gorilla/mux"
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/database/cosmosdb"
+	"github.com/Azure/ARO-RP/pkg/frontend/middleware"
+	"github.com/Azure/ARO-RP/pkg/util/restconfig"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	OVSFlowContianerName string = "container-00"
+	OVSFlowPodName       string = "aro-ovsflows"
+	OVSFlowPodNamespace  string = "default"
+)
+
+func (f *frontend) getOVSFlows(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	log := ctx.Value(middleware.ContextKeyLog).(*logrus.Entry)
+
+	config, err := f.getKubernetesClientSet(ctx, r, log)
+	if err != nil {
+		log.Error(err)
+	}
+
+	client := Client{config}
+	// Main logic written in f._getOSVFlow() func
+	b, err := client._getOVSFlows(ctx, r, log, config)
+	if err != nil {
+		log.Error(err)
+	}
+	reply(log, w, nil, b, err)
+}
+
+func (client Client) _getOVSFlows(ctx context.Context, r *http.Request, log *logrus.Entry, config *kubernetes.Clientset) ([]byte, error) {
+	//Checking if the node mentioned in the query is correct or not.
+	nodeName := r.URL.Query().Get("node")
+	if nodeName == "" {
+		err := api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, "", "Invalid Node Name")
+		return nil, err
+	}
+
+	// Checking if the given node exists
+	node, err := config.CoreV1().Nodes().Get(ctx, nodeName, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	// Creating a pod which will run as privileged in the given node to execute "$ ovs-ofctl -O OpenFlow13 dump-flows br0" on that node to extract the OVS flows.
+	pod, err := client.createOVSFlowPod(ctx, node.Name, log)
+	if err != nil {
+		log.Error(err)
+		return nil, err
+	}
+
+	//  The below loop waits of the phase of the created pod to be Succeeded, if not it times out after 60 seconds and returns error.
+	for i := 0; pod.Status.Phase != corev1.PodSucceeded; i++ {
+		if i == 60 {
+			err = api.NewCloudError(http.StatusNotFound, api.CloudErrorCodeInternalServerError, "", "")
+			return nil, err
+		}
+		pod, err = config.CoreV1().Pods(OVSFlowPodNamespace).Get(ctx, pod.Name, metav1.GetOptions{})
+		if err != nil {
+			log.Error(err)
+			return nil, err
+		}
+		time.Sleep(time.Second)
+	}
```

Comment:
I still think that we should look into using `k8s.io/apimachinery/pkg/util/wait` as suggested [previously](https://github.com/Azure/ARO-RP/pull/1859#discussion_r760130380). Usually `time.Sleep` causes a loto of pain in tests.

Reply:
Yes I looked into it ..and will try to look include that here ..